scripts/repo-build-web

```diff
@@ -57,77 +59,185 @@ cat > "$repodir/index.html" << WEBS
+
+            .listing-name, .listing-version, .listing-license {
+                width: 15%;
+            }
+
+            .listing-desc {
+                width: 45%;
+            }
+
+            .sortable th {
+                cursor: pointer;
+            }
+
+            .sortable th:not(.sort-asc):not(.sort-desc)::after {
+                content: " ⇅";
+            }
+
+            .sortable th.sort-asc::after {
+                content: " ↓";
+            }
+
+            .sortable th.sort-desc::after {
+                content: " ↑";
+            }
         </style>
     </head>
     <body>
         <a href="../">Back to Repository Home Page</a>
         <h1>Toltec Package Listing</h1>
 
-        <table>
+WEBS
+
+section "Indexing sections"
+declare -A pkgsections
+
+for recipedir in "$recipesdir"/*; do
+    section="$(
+        load-recipe "$recipedir"
+        echo "$section"
+    )"
+    pkgsections["$section"]=1
+done
+
+for cursection in "${!pkgsections[@]}"; do
+    section "Making $cursection index"
+
+    cat >> "$indexfile" << WEBS
+        <h2>$cursection</h2>
+
+        <table class="listing sortable">
             <thead>
                 <tr>
                     <th>Name</th>
                     <th>Description</th>
                     <th>Version</th>
-                    <th>Category</th>
                     <th>License</th>
                 </tr>
             </thead>
+
+            <colgroup>
+                <col class="listing-name">
+                <col class="listing-desc">
+                <col class="listing-version">
+                <col class="listing-license">
+            </colgroup>
 WEBS
 
-# Build packages index
-section "Making packages index"
+    for recipedir in "$recipesdir"/*; do
+        (
+            load-recipe "$recipedir"
 
-for recipedir in "$recipesdir"/*; do
-    (
-        load-recipe "$recipedir"
-        pkgid="$(package-id)"
+            if [[ $section = "$cursection" ]]; then
+                pkgid="$(package-id)"
 
-        status "Adding entry for $pkgid"
+                status "Adding entry for $pkgid"
 
-        if [[ -z $url ]]; then
-            href="$url"
-        else
-            href="<a href=\"$url\">$pkgname</a>"
-        fi
+                if [[ -z $url ]]; then
+                    href="$url"
+                else
+                    href="<a href=\"$url\">$pkgname</a>"
+                fi
 
-        cat >> "$repodir/index.html" << WEBS
+                cat >> "$indexfile" << WEBS
             <tr>
                 <td>$href</td>
                 <td>$pkgdesc</td>
                 <td>$pkgver</td>
-                <td>$section</td>
                 <td><a href='https://spdx.org/licenses/$license.html'>$license</a></td>
             </tr>
 WEBS
-    )
+            fi
+        )
+    done
 
+    cat >> "$indexfile" << 'WEBS'
+        </table>
+WEBS
 done
 
-cat >> "$repodir/index.html" << WEBS
-        </table>
+cat >> "$indexfile" << 'WEBS'
+        <script>
```

Comment:
why is this indented?

Reply:
This is to match the indent level of the rest of the generated HTML output.